src/main/java/ch/ibw/appl/tudu/server/item/service/ItemService.java

```diff
@@ -66,6 +67,34 @@ public Item deleteById(long requestedId) {
         return matches;
     }
 
+    public List<Item> getByUserId(Long requestedUserId) {
+        List<Item> matches = new ArrayList<>();
+
+        for (Item item : items) {
+            if (item.userId == requestedUserId) {
+                matches.add(item);
+            }
+        }
+        return matches;
+    }
+
+    public List<Item> geByUserIdAndFilter(long requestedUserId, String filter) {
+        String[] keyValue = filter.split(":");
+        List<Item> matches = new ArrayList<>();
+
+        if (keyValue[0].equalsIgnoreCase("description")) {
```

Comment:
diesen filter if/for loop hast du jetzt bereits zweimal oder? hier hast du ihn nun noch erweiter um die requestedUserId. Das ist nicht optimal. so kannst du den filter if/for loop code nicht mehr in eine methode auslagern.

Reply:
verstehe ich richtig,  weil ich den Code doppelt habe macht es Sinn diesen in einer methode auszulagern, aber diesen habe ich bereits. ich könnte den loop in eine methode auslagern oder ich nutze einfach die bereits vorhandene getByFilter() Methode und erstelle eine "Zwischen-Liste" a la: 

List<Items> matchesFilter = getByFilter(filter);
for (Item item : matchesFilter) 
   if (item.userId.equals(requestedUserId) {
      matches.add(items)
   }
}
return matches

Reply:
so hatte ich mir das auch vorgestellt 👍 .
alernative hättest du auch getByFilter(filter + ";id:"+requestedUserId) machen können und getByFilter erweitern, dass er auch nach id feld filtern kann.